addons/udes_delivery_control/models/picking_extra_fields.py

```diff
@@ -0,0 +1,146 @@
+from odoo import api, fields, models, _
+from ..misc import date_diff, validate_dates
+
+
+class StockPickingDeliveryControl(models.Model):
+    """
+        Model to hold all extra fields of stock.picking object
+    """
+
+    _name = "udes_picking_extras"
+    _description = "UDES Picking Extra Fields"
+
+    picking_id = fields.Many2one(
+        "stock.picking", required=True, index=True, copy=False, ondelete="cascade"
+    )
+
+    # Unloading Fields
+    u_user_id = fields.Many2one("res.users", string="User", help="Picking completed by user")
+    u_delivery_note = fields.Char("Delivery Note")
+    u_location_id = fields.Many2one("stock.location", string="Location")
+    u_vehicle_scheduled_date = fields.Datetime(
+        "Scheduled Date/Time", help="Date/Time when the vehicle is expected to arrive"
+    )
+    u_vehicle_arrival_date = fields.Datetime("Arrival Date/Time", help="Arrival time of Vehicle")
+    u_week = fields.Integer("Week", compute="_compute_u_week", store=True, default=0)
+    u_unloading_start_date = fields.Datetime("Date Start", help="Unloading start date/time")
+    u_unloading_end_date = fields.Datetime("Date End", help="Unloading completed date/time")
+    u_time_taken = fields.Float(
+        "Time Taken HH:MM",
+        compute="_compute_u_time_taken",
+        store=True,
+        readonly=True,
+        help="The amount of time taken to complete unloading in HH:MM format",
+    )
+    u_vehicle_type = fields.Char("Vehicle Type")
```

Comment:
should this be a selection field otherwise we might get inconsistent types

Reply:
Yeah inconsistent types would be a big pain for reporting. :)

Worth checking with product. If this Vehicle Type field needs to be quite specific then it might be better having a separate entity for this even, so that options can be managed through the UI.

Reply:
I checked with product team and they said it would be good idea to have a selection field with a menu to configure it. 
Only inbound user to have permission to add, remove and edit vehicle type

Reply:
This is now implemented and can be reviewed